src/os/solaris/solaris_sigar.c

```diff
@@ -263,6 +263,11 @@ static int zone_mem_get(sigar_t *sigar, sigar_mem_t *mem)
     return SIGAR_OK;
 }
 
+char * sigar_get_machine_id(void) {
+  char machine_id[256] = "NOTIMPLEMENTED";
```

Comment:
these need to be 'static' to avoid returning stack garbage, though if you make machine_id part of sigar_sys_info_get, that might solve the problem too.